src/target/riscv/program.c

```diff
@@ -45,6 +45,8 @@ int riscv_program_write(struct riscv_program *program)
 int riscv_program_exec(struct riscv_program *p, struct target *t)
 {
 	keep_alive();
+	if (openocd_is_shutdown_pending())
+		return ERROR_SERVER_REMOTE_CLOSED;
```

Comment:
I can see three cases in this merge request where the error code is different than `ERROR_SERVER_INTERRUPTED`. Are these intentional or just typos?

```suggestion
	if (openocd_is_shutdown_pending())
		return ERROR_SERVER_INTERRUPTED;
```

Reply:
These are just typos. Thanks for the catch! Addressed.